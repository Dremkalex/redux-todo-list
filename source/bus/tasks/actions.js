// Types
import { types } from './types';

export const tasksActions = {
    // Sync
    fillTasks: (tasks) => ({
        type:    types.FILL_TASKS,
        payload: tasks,
    }),

    createTask: (task) => ({
        type:    types.CREATE_TASK,
        payload: task,
    }),

    removeTask: (taskId) => ({
        type:    types.REMOVE_TASK,
        payload: taskId,
    }),

    updateTask: (updatedTask) => ({
        type:    types.UPDATE_TASK,
        payload: updatedTask,
    }),

    updateNewTaskMessage: (message) => ({
        type:    types.UPDATE_NEW_TASK_MESSAGE,
        payload: message,
    }),

    updateEditTaskMessage: (message) => ({
        type:    types.UPDATE_EDIT_TASK_MESSAGE,
        payload: message,
    }),

    updateTasksFilter: (message) => ({
        type:    types.UPDATE_TASKS_FILTER,
        payload: message,
    }),

    setEditTaskId: (editTaskId) => ({
        type:    types.SET_EDIT_TASK_ID,
        payload: editTaskId,
    }),

    resetEditTaskId: () => ({
        type: types.RESET_EDIT_TASK_ID,
    }),

    // Async
    fetchTasksAsync: () => ({
        type: types.FETCH_TASKS_ASYNC,
    }),

    createTaskAsync: (newTaskMessage) => ({
        type:    types.CREATE_TASK_ASYNC,
        payload: newTaskMessage,
    }),

    removeTaskAsync: (taskId) => ({
        type:    types.REMOVE_TASK_ASYNC,
        payload: taskId,
    }),

    updateTaskAsync: (updatedTask) => ({
        type:    types.UPDATE_TASK_ASYNC,
        payload: updatedTask,
    }),

    completeAllTasksAsync: (tasks) => ({
        type:    types.COMPLETE_ALL_TASK_ASYNC,
        payload: tasks,
    }),
};
