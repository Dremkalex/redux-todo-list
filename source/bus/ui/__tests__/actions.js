// Actions
import { uiActions } from '../actions';

describe('ui actions', () => {
    test('start fetching', () => {
        expect(uiActions.startFetching()).toMatchSnapshot();
    });

    test('stop fetching', () => {
        expect(uiActions.stopFetching()).toMatchSnapshot();
    });

    test('emit error', () => {
        expect(uiActions.emitError(__.error)).toMatchSnapshot();
    });
});
