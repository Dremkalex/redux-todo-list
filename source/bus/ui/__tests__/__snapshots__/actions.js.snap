// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui actions emit error 1`] = `
Object {
  "error": true,
  "meta": null,
  "payload": [Error: TEST_ERROR_MESSAGE.],
  "type": "EMIT_ERROR",
}
`;

exports[`ui actions start fetching 1`] = `
Object {
  "type": "START_FETCHING",
}
`;

exports[`ui actions stop fetching 1`] = `
Object {
  "type": "STOP_FETCHING",
}
`;
